File: app/src/main/java/com/abishov/hexocat/home/trending/TrendingContract.java
package com.abishov.hexocat.home.trending;

import com.abishov.hexocat.github.filters.SearchQuery;
import io.reactivex.Observable;
import io.reactivex.functions.Consumer;

interface TrendingContract {

  interface View extends com.abishov.hexocat.common.views.View {

    Observable<SearchQuery> searchQueries();

    Consumer<TrendingViewState> bindTo();
  }

  interface Presenter extends com.abishov.hexocat.common.views.Presenter<View> {

  }
}

File: app/src/main/java/com/abishov/hexocat/home/trending/TrendingContract.kt
package com.abishov.hexocat.home.trending

import com.abishov.hexocat.github.filters.SearchQuery
import io.reactivex.Observable
import io.reactivex.functions.Consumer

internal interface TrendingContract {

  interface View : com.abishov.hexocat.common.views.View {

    fun searchQueries(): Observable<SearchQuery>

    fun bindTo(): Consumer<TrendingViewState>
  }

  interface Presenter : com.abishov.hexocat.common.views.Presenter<View>
}
